golem-api-grpc/proto/golem/worker/invocation_context.proto

```diff
@@ -2,10 +2,41 @@ syntax = "proto3";
 
 package golem.worker;
 
+import "google/protobuf/timestamp.proto";
 import "golem/worker/worker_id.proto";
 
 message InvocationContext {
   golem.worker.WorkerId parent = 1;
   repeated string args = 3;
   map<string, string> env = 4;
+  optional TracingInvocationContext tracing = 5;
+}
+
+message TracingInvocationContext {
+  string trace_id = 1;               // hexadecimal representation of a 128-bit trace ID
+  repeated InvocationSpan spans = 2; // stack of spans, first item is the current, last one is the root
+  repeated string trace_state = 3;
+}
+
+message InvocationSpan {
+  oneof span {
+    LocalInvocationSpan local = 1;
+    ExternalParentSpan external_parent = 2;
+  }
+}
+
+message LocalInvocationSpan {
+  uint64 span_id = 1;
+  google.protobuf.Timestamp start = 2;
```

Comment:
do we need an end_time here?

Reply:
Turned out that I'm never keeping finished spans in memory so no. Also in the protocol, even when multiple spans are ongoing, on invocation boundaries only the "current stack" is transmitted, which consist of open spans.

Reply:
I meant for eventual oplog processors. I think often you will need to report endtime to your telemetry backends. Or are plugins supposed to infer the end time by looking for the matching invocation-end entry?

As another question, how are we going to do attributes on the current span if we only persist the invocations? Or are these https://www.notion.so/golemcloud/Invocation-Context-implementation-19ea4cb355ec80a3bc69eefcc21926c4?pvs=4#19ea4cb355ec800bbe5af693dd9c74ea coming as dedicated oplog entries in another pr?